client-go/parser/perms.go

```diff
@@ -0,0 +1,129 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/deis/deis/client-go/cmd"
+	docopt "github.com/docopt/docopt-go"
+)
+
+// Perms routes perms commands to their specific function.
+func Perms(argv []string) error {
+	usage := `
+Valid commands for perms:
+
+perms:list            list permissions granted on an app
+perms:create          create a new permission for a user
+perms:delete          delete a permission for a user
+
+Use 'deis help perms:[command]' to learn more.
+`
+	if len(argv) < 2 {
+		return permsList([]string{"perms:list"})
+	}
+
+	switch argv[1] {
+	case "list":
+		return permsList(combineCommand(argv))
+	case "create":
+		return permCreate(combineCommand(argv))
+	case "delete":
+		return permDelete(combineCommand(argv))
+	case "--help":
+		fmt.Print(usage)
+		return nil
+	default:
+		PrintUsage()
+		return nil
+	}
+}
+
+func permsList(argv []string) error {
+	usage := `
+Lists all users with permission to use an app, or lists all users with system
+administrator privileges.
+
+Usage: deis perms:list [-a --app=<app>|--admin]
+
+Options:
+  -a --app=<app>
+    lists all users with permission to <app>. <app> is the uniquely identifiable name
+    for the application.
+  --admin
+    lists all users with system administrator privileges.
+`
+
+	args, err := docopt.Parse(usage, argv, true, "", false, true)
+
+	if err != nil {
+		return err
+	}
+
+	admin := args["--admin"].(bool)
+
+	return cmd.PermsList(safeGetValue(args, "--app"), admin)
+}
+
+func permCreate(argv []string) error {
```

Comment:
For my code I went with singular versions when dealing with a single object (creating, removing, getting) and plural with commands that deal with multiple objects (listing). I'm happy to change if it is confusing, but is probably a client-go wide change.